src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -901,6 +892,29 @@ def _find_independent_spans(self, spans: list[Span]) -> list[Span]:
                 independent_spans.append(span)
         return independent_spans
 
+    def _calculate_cost(self, independent_spans: list[Span]) -> float:
+        """
+        Calculates the cost saved by running spans in parallel,
+        this is the maximum time saved of running all independent queries in parallel
+        note, maximum means it does not account for db connection times and overhead associated with parallelization,
+        this is where thresholds come in
+        """
+        consecutive_spans = self.consecutive_db_spans
+        consecutive_total_duration = self._sum_span_duration(consecutive_spans)
+        parallel_total_duration = consecutive_total_duration
+        independent_span_durations = [
+            get_span_duration(span).total_seconds() * 1000 for span in independent_spans
+        ]
+        max_independent_span_duration = max(independent_span_durations)
+
+        for span_duration in independent_span_durations:
+            parallel_total_duration -= span_duration
+
+        if parallel_total_duration < max_independent_span_duration:
+            parallel_total_duration = max_independent_span_duration
+
+        return consecutive_total_duration - parallel_total_duration
+
```

Comment:
This algorithm seems a bit more complicated than it needs to be and it's a little tricky to read imo. I believe that if we did it this way instead, it would greatly simplify our code:

- Take the end timestamp of the longest span in the sequence and then the end timestamp of the very last span in the sequence. This would represent the time shaved off if we were to parallelize the calls

Diagram for example:

```
(longest span)
[      span1      ]
                  [ span2 ]
                          [ span3 ]

| -------- total duration ------- |

Parallelized:

[      span1      ]
[ span2 ]
[ span3 ]
                  | ------------- |
                  (time shaved off)
                           


```

Reply:
Feel free to correct me if I'm misunderstanding or missing details though

Reply:
I don’t believe that method will work, because we’re not making every span in the consecutive set of span in parallel, just a few.
So it would be something like
[      span1      ]
                  [ span2 ]
                          [ span3 ]
Parallel
[      span1      ]
[ span2 ]                  
                   [ span3 ]

Reply:
I pushed a change similar to this algorithm, it simplifies the code more and addresses the comments we discussed, thanks for the help!